File: skills/write-readable-code/assets/layout/dotnet/template.fs
// F# file template: namespace, opens, literals, types with members, then
// internal modules. F# compiles top to bottom, so helpers precede users.
namespace Project.Component

open System

[<AutoOpen>]
module internal Internal =
    [<Literal>]
    let AssemblyConstant = 2

    let helper (value: int) = Math.Abs value

type PublicType() =
    member _.PublicMember() = helper AssemblyConstant
    member internal _.AssemblyMember() = AssemblyConstant
    member private _.PrivateMember() = ()

type internal InternalType() =
    member _.Run() = helper -1

File: skills/write-readable-code/assets/layout/dotnet/template.cs
// C# file template: usings, file-scoped namespace, then one main type with
// its members together; internal helpers after the public surface.
using System;

namespace Project.Component;

public static class Constants
{
    public const int PublicValue = 1;
    internal const int AssemblyValue = 2;
}

public sealed class PublicType
{
    private int state;

    public PublicType() => state = Constants.AssemblyValue;

    public int PublicMethod() => PrivateMethod() + Constants.PublicValue;

    internal int AssemblyMethod() => state;

    private int PrivateMethod() => InternalType.Helper(state);
}

// C# has no private top-level types; the narrowest top-level scope is
// internal (or file-local `file` types since C# 11).
internal static class InternalType
{
    internal static int Helper(int value) => Math.Abs(value);
}
